sros/docker/launch.py

```diff
@@ -827,10 +826,22 @@ def __init__(self, hostname, username, password, mode, variant_name, conn_mode):
             # LC VM Instantiation
             start_eth = 1
             lc_slot_tracker = []
+
             for lc in variant["lcs"]:
-                lc_slot = lc["slot"]
+                lc_slot = lc.get("slot", None)
+
+                # Try to retrive from timos_line
+                if not lc_slot:
+                    timos_line = lc.get("timos_line")
+                    lc = parse_variant_line(timos_line, lc)
+                    self.logger.info(f"New reconsructed LC: {lc}")
```

Comment:
why don't we add the logic of parsing the variant line to line https://github.com/hellt/vrnetlab/blob/428ee4f70a887f8e3b438da39a9041275d7513aa/sros/docker/launch.py#L742

There we get the variant obj from the embedded variants, and after that we parse the variant lines for the retrieved object.
That way, we keep this code block of LC VM instantiation intact

Reply:
Do yo mean to update the all variant LC information before entering the LC VM instantiation code block?
I think that is doable. I will push again today for that changes.